tests/test_diffusers.py

```diff
@@ -2929,6 +2929,95 @@ def test_dreambooth_lora_sdxl(self):
+                --dataset_name {tmpdir}
+                --resolution 256
+                --train_batch_size 1
+                --gradient_accumulation_steps 1
+                --max_train_steps 20
+                --rank 4
+                --learning_rate 1e-04
+                --max_grad_norm 1
+                --lr_scheduler constant
+                --lr_warmup_steps 0
+                --gaudi_config_name Habana/stable-diffusion
+                --use_hpu_graphs_for_training
+                --use_hpu_graphs_for_inference
+                --mixed_precision bf16
+                --bf16
+                --sdp_on_bf16
+                --num_validation_images 1
+                --output_dir {tmpdir}
+                """.split()
+            if train_text_encoder:
+                test_args.append("--train_text_encoder")
+            test_args.append("--instance_prompt")
+            test_args.append(instance_prompt)
+            test_args.append("--validation_prompt")
+            test_args.append(validation_prompt)
+            p = subprocess.Popen(test_args)
+            return_code = p.wait()
+
+            # Ensure the run finished without any issue
+            self.assertEqual(return_code, 0)
+            # save_pretrained smoke test
+            self.assertTrue(os.path.isfile(os.path.join(tmpdir, "pytorch_lora_weights.safetensors")))
+
+            # make sure the state_dict has the correct naming in the parameters.
+            lora_state_dict = safetensors.torch.load_file(os.path.join(tmpdir, "pytorch_lora_weights.safetensors"))
+            is_lora = all("lora" in k for k in lora_state_dict.keys())
+            self.assertTrue(is_lora)
+
+            # when not training the text encoder, all the parameters in the state dict should start
+            # with `"transformer"` in their names.
+            if train_text_encoder:
+                starts_with_transformer = all(
+                    k.startswith("transformer") or k.startswith("text_encoder") or k.startswith("text_encoder_2")
+                    for k in lora_state_dict.keys()
+                )
+            else:
+                starts_with_transformer = all(key.startswith("transformer") for key in lora_state_dict.keys())
+            self.assertTrue(starts_with_transformer)
+
+            if cache_dir.is_dir():
+                shutil.rmtree(cache_dir)
+
+    @check_gated_model_access("stabilityai/stable-diffusion-3-medium-diffusers")
+    @pytest.mark.skipif(IS_GAUDI1, reason="does not fit into Gaudi1 memory")
+    def test_dreambooth_lora_sd3(self):
+        orig_value = os.environ.get("PT_HPU_MAX_COMPOUND_OP_SIZE")
+        os.environ["PT_HPU_MAX_COMPOUND_OP_SIZE"] = "1"
+        try:
+            self._test_dreambooth_lora_sd3(train_text_encoder=False)
+        finally:
+            if orig_value is not None:
+                os.environ["PT_HPU_MAX_COMPOUND_OP_SIZE"] = orig_value
+            else:
+                del os.environ["PT_HPU_MAX_COMPOUND_OP_SIZE"]
```

Comment:
Can we also add a short test for `train_text_to_image_sd3.py`? Just doing a couple of training iterations for instance, to check the script runs without error.

Reply:
Done